src/AMSlib/wf/basedb.hpp

```diff
@@ -1801,25 +1999,78 @@ class RMQPublisher
    */
   void stop() { event_base_loopexit(_loop.get(), NULL); }
 
-  void release_messages() { _handler->release_message_buffers(); }
+  /**
+   * @brief Check if the underlying connection has no errors
+   * @return True if no errors
+   */
+  bool connection_valid() { return _handler->connection_valid(); }
 
-  void publish(AMSMessage&& message) { _handler->publish(std::move(message)); }
+  /**
+   * @brief Return the messages that have not been acknolwged.
+   * It does not mean they have not been delivered but the
+   * acknowledgements have not arrived yet.
+   * @return A map of messages (ID, data)
+   */
+  std::vector<AMSMessage> get_buffer_msgs()
+  {
+    return std::move(_handler->internal_msg_buffer());
```

Comment:
Do we want to use move here? Are we certain that the `_handler` class will never access the data_ptrs vector?  My understanding is what you do is correct, but it can be confusing to use.

Reply:
The idea was that if you ask for the internal message buffer that's because you cannot recover the connection so you can move the vector anyway (which leave the object in an unknown state but you don't care as you destroy it).

However, I agree it could backfire. I removed the move. I am planning to encapsulate the vector of AMSMessage and the mutex into an object that will represent the state of `RMQPublisher`  and that will take care of manipulating the underlying buffer of messages.